File: src/python/getbalance.py
import os
from dotenv import load_dotenv
from web3 import Web3

# Load environment variables
load_dotenv()
SEPOLIA_URL = os.getenv("Sepolia_Url")
FLASHLOAN_ARBITRAGE_CONTRACT = os.getenv("Flashloan_Arbitrage")
USDC_ADDRESS = os.getenv("Sepolia_USDC")
DAI_ADDRESS = os.getenv("Sepolia_DAI")

# Connect to Sepolia network
web3 = Web3(Web3.HTTPProvider(SEPOLIA_URL))
if not web3.is_connected():
    print("Failed to connect to Sepolia network")
    exit()

# ERC20 ABI for balanceOf function
ERC20_ABI = [
    {
        "constant": True,
        "inputs": [{"name": "_owner", "type": "address"}],
        "name": "balanceOf",
        "outputs": [{"name": "", "type": "uint256"}],
        "type": "function"
    }
]

def check_balance(token_address):
    try:
        # Load token contract
        token_contract = web3.eth.contract(address=token_address, abi=ERC20_ABI)

        # Call balanceOf function
        balance = token_contract.functions.balanceOf(
            FLASHLOAN_ARBITRAGE_CONTRACT
        ).call()
        print(f"Balance of token {token_address}: {balance}")
    except Exception as e:
        print(f"Error in checking balance: {e}")

# Check balances for USDC and DAI
check_balance(USDC_ADDRESS)
check_balance(DAI_ADDRESS)

File: src/javascript/getbalance.js
require('dotenv').config(); // Load environment variables
const { Web3 } = require('web3'); // Web3 setup

// Load environment variables
const SEPOLIA_URL = process.env.Sepolia_Url; // Sepolia network URL
const FLASHLOAN_ARBITRAGE_CONTRACT = process.env.Flashloan_Arbitrage; // Flashloan Arbitrage Smart Contract

// ERC20 ABI for checking balance
const ERC20_ABI = [
    {
        "constant": true,
        "inputs": [{ "internalType": "address", "name": "_owner", "type": "address" }],
        "name": "balanceOf",
        "outputs": [{ "internalType": "uint256", "name": "", "type": "uint256" }],
        "stateMutability": "view",
        "type": "function"
    }
];

// Connect to the Sepolia network
const web3 = new Web3(SEPOLIA_URL);

// Function to check token balance
async function checkBalance(tokenAddress) {
    try {
        const tokenContract = new web3.eth.Contract(ERC20_ABI, tokenAddress);
        const balance = await tokenContract.methods.balanceOf(FLASHLOAN_ARBITRAGE_CONTRACT).call();
        console.log(`Balance of token ${tokenAddress} in Flashloan Arbitrage Contract ${FLASHLOAN_ARBITRAGE_CONTRACT}: ${balance}`);
    } catch (error) {
        console.error("Error checking balance:", error.message);
    }
}

// Example usage: Check USDC and DAI balances in the Flashloan Arbitrage contract
const USDC_ADDRESS = process.env.Sepolia_USDC;
const DAI_ADDRESS = process.env.Sepolia_DAI;

checkBalance(USDC_ADDRESS);
checkBalance(DAI_ADDRESS);
